source/parser/comptime.civet

```diff
@@ -74,6 +74,12 @@ function serialize(value: ???): string
         if /\{\s+\[native code]\s+\}$/.test string
           // builtin, or returned from Function::bind
           throw new TypeError "cannot serialize native function"
+        // Check for ES6 methods that have been removed from the object, and add "function"
+        unless /^(?:async[\s\n]+)?(?:function|\(|\w+[\n\s]*=>)/u.test string
```

Comment:
```suggestion
        unless /^(?:async[\s\n]+)?(?:function(?!\p{ID_Continue})|\(|(?:\p{ID_Start}|[_$])(?:\p{ID_Continue}|[\u200C\u200D$])*\s*=>)/u.test string
```

* Don't want to match `functionfoo()`
* Copy regexp from `IdentifierName` to properly handle other leading identifiers. Particularly crucial is `$` which is generated by `&`. (Alternatively, we could probably just use `[^()]*=>`...)
* `\n` is included in `\s` so not sure why you're including it explicitly.

Reply:
My implementation already handles `$ => $` correctly which is probably a result of me making the same mistake twice: `replace` is called but no replacement is performed. I seem to remember some regex implementation not counting linebreaks in `\s`, but I can't find any where that's the case, so I was probably just mistaken. I'll update this and the other regex accordingly.

Reply:
Actually, I think this can be done with a single regex -- the second one is unnecessary, and the first one can be passed directly to `replace` after removing the first `?:`.